tests/console/nvme_checks.pm

```diff
@@ -0,0 +1,116 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Smoke tests for NVM Express Controller.
+#
+# - Install nvme-cli which is a tool to manage NVM Express
+# - Check sys path that should be hold details
+# - Check if we have any NVMe controllers on the PCI bus
+# - Check if we have NVMe devices
+# - Check if we have namespace
+# - Send Identify Admin Command to the NVMe controller
+# - Issue a Read command both using the helper and manually
+# - Compare it to what we have with a regular dd
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base "installbasetest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use scheduler 'get_test_suite_data';
+
+sub _smoke_checks {
+    my $nvm_test_data = shift;
+    assert_script_run("lspci -nn | grep -i nvm");
+    assert_script_run("test -d /sys/class/nvme");
+    assert_script_run("test -c /dev/nvme || echo 'nvm character devices not found'");
+
+    my @nvm_partitions = split(/\n/, script_output("ls -l /dev/$nvm_test_data->{nvm_disk}*"));
+    foreach (@nvm_partitions) {
+        assert_script_run("test -b $_ || echo 'nvm is not a block device'");
+    }
+}
+
+sub _exercise_nvme_commands_and_validate_output {
+    my $nvm_test_data = shift;
+    # send Identify Admin Command to the NVMe controller.
+    assert_script_run("nvme id-ctrl -H /dev/$nvm_test_data->{nvm_char_device}");
+    save_screenshot;
```

Comment:
instead of a screenshot it normally helps a lot to have the output displayed in the box in openQA with script_output or validate_script_output. I'm not sure with script_run, you might need to redirect it, right? could you check how could be easily visualized?

Reply:
make sense. i will try

Reply:
added